src/cleanfigure.m

```diff
@@ -367,6 +368,259 @@ function coarsenLine(meta, handle, minimumPointsDistance)
   return;
 end
 % =========================================================================
+function simplifyLine(meta, handle, targetResolution)
+  % Reduce the number of data points in the line handle by removing
+  % points which add features with area smaller than 1/4 of a pixel at the
+  % target resolution. 
+  %
+  % targetResolution is in format 'WxH@Res', eg. '15x9@600' for a 15cm by 9cm
+  % figure at 600 ppcm.
+  % 
+  % Use targetResolution = 'off' to disable line simplification
+
+  % Extract the data from the current line handle.
+  xData = get(handle, 'XData');
+  yData = get(handle, 'YData');
+  zData = get(handle, 'ZData');
+
+  if ~isempty(zData)
+    % Don't simplify 3d plots
+    return;
+  end
+  if isempty(xData) || isempty(yData)
+      return;
+  end
+  if numel(xData) <= 2
+      return;
+  end
+  if strcmpi(targetResolution,'off')
```

Comment:
Wouldn't it make more sense to have the algorithm switched off if the resolution (`param(3)`) is infinity or NaN?